src/components/NewDatePicker/index.native.js

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import {View, Keyboard} from 'react-native';
+import moment from 'moment';
+import _ from 'underscore';
+import TextInput from '../TextInput';
+import CONST from '../../CONST';
+import styles from '../../styles/styles';
+import {propTypes, defaultProps} from './datePickerPropTypes';
+import withKeyboardState, {keyboardStatePropTypes} from '../withKeyboardState';
+import CalendarPicker from '../CalendarPicker';
+
+const datepickerPropTypes = {
+    ...propTypes,
+    ...keyboardStatePropTypes,
+};
+
+const datePickerDefaultProps = {
+    ...defaultProps,
+};
+
+class NewDatePicker extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isPickerVisible: false,
+            selectedDate: moment(props.value || props.defaultValue).toDate(),
+        };
+
+        this.showPicker = this.showPicker.bind(this);
+        this.reset = this.reset.bind(this);
+        this.hidePicker = this.hidePicker.bind(this);
+        this.setDate = this.setDate.bind(this);
+    }
+
+    componentDidMount() {
+        if (!this.props.autoFocus) {
+            return;
+        }
+        this.showPicker();
+    }
+
+    /**
+     * Trigger the `onInputChange` handler when the user input has a complete date or is cleared
+     * @param {Date} selectedDate
+     */
+    setDate(selectedDate) {
+        this.setState({selectedDate}, () => {
+            this.props.onInputChange(moment(selectedDate).format(CONST.DATE.MOMENT_FORMAT_STRING));
+            this.hidePicker();
+        });
+    }
+
+    /**
+     * displays calendar picker on the screen
+     */
+    showPicker() {
+        this.initialValue = this.state.selectedDate;
```

Comment:
I think we no longer need to store the `initialValue` as we are not longer using the native picker, right?

Reply:
Yes, also reset function is redundant. Removing it